### PR TITLE
[Python] `fetchmany` implemented

### PR DESCRIPTION
**Pull request by Tishj**
This PR implements `fetchmany` in the DuckDB Python API, according to the DB API 2.0 specification:
https://peps.python.org/pep-0249/#fetchmany

There is a slight difference between this implementation and the specification though.
The specification says an empty sequence (`[]`) should be returned to indicate the result is closed, so:
```py
res = cursor.execute("select i from range(4) tbl(i)");
>> cursor.fetchmany(3)
[(0,), (1,), (2,)]
>> cursor.fetchmany(3)
[(3,)]
>> cursor.fetchmany(3)
[]
>> cursor.fetchmany(3)
# Error
```

But this implements the following behavior:
```py
res = cursor.execute("select i from range(4) tbl(i)");
>> cursor.fetchmany(3)
[(0,), (1,), (2,)]
>> cursor.fetchmany(3)
[(3,)]
>> cursor.fetchmany(3)
# Error
```

I believe it requires some extra changes in the internal `QueryResult::Fetch` methods to match this behavior, so I'm hesitant to implement it at this time.

### CODE REVIEW

**Review by Mause — CHANGES_REQUESTED** [review]
We need to match the spec properly, otherwise people will just have to handle the issues downstream. Can we not catch the error you mentioned?

**Comment by Tishj** [comment]
Yea I figured that was going to be the answer, I just tested sqlite3's behavior on this, and it seems they never throw an error, so your solution is possible, but it's not very clean.

I'm currently trying to get more control over the internal result and only "close" it when we get an empty result

File: tools/pythonpkg/src/pyresult.cpp [hunk 1]
```diff
@@ -203,6 +205,22 @@ py::object DuckDBPyResult::Fetchone() {
...
+py::list DuckDBPyResult::Fetchmany(idx_t size) {
+	py::list res;
+	for (idx_t i = 0; i < size; i++) {
+		try {
+			auto fres = Fetchone();
+			if (fres.is_none()) {
+				break;
+			}
+			res.append(fres);
+		} catch (InvalidInputException &e) {
```
Comment (Mause) [inline tools/pythonpkg/src/pyresult.cpp:217]: Hmm, quite a broad catch. Is it worth checking the error message?

File: tools/pythonpkg/tests/fast/api/test_dbapi00.py [hunk 2]
```diff
@@ -4,11 +4,63 @@
...
+            if (len(res) == 0):
+                break
+        assert(iteration_count == expected_iteration_count)
+        assert(len(list_of_results) == truth_value)
+        assert_result_equal(list_of_results)
+
+    def test_fetchmany_too_many(self, duckdb_cursor):
+        truth_value = len(duckdb_cursor.execute('select * from integers').fetchall())
+        duckdb_cursor.execute('select * from integers')
+        list_of_results = []
```
Comment (Mause) [inline tools/pythonpkg/tests/fast/api/test_dbapi00.py:60]: nitpick: unused variable?

**Review by Mause — CHANGES_REQUESTED** [review]

File: tools/pythonpkg/src/pyresult.cpp (hunk 1 above)
Comment (Tishj) [reply tools/pythonpkg/src/pyresult.cpp]: Fixed it to not need a try-catch

**Pull request merged**